src/NuGet.Core/NuGet.Frameworks/FrameworkRuntimePair.cs

```diff
@@ -1,12 +1,18 @@
 ﻿using System;
+using System.Globalization;
 
 namespace NuGet.Frameworks
 {
-    public class FrameworkRuntimePair : IEquatable<FrameworkRuntimePair>, IComparable<FrameworkRuntimePair>
+#if NUGET_FRAMEWORKS_INTERNAL
+    internal
+#else
+    public
+#endif
+    class FrameworkRuntimePair : IEquatable<FrameworkRuntimePair>, IComparable<FrameworkRuntimePair>
     {
-        public NuGetFramework Framework { get; }
-        public string RuntimeIdentifier { get; }
-        public string Name { get; }
+        public NuGetFramework Framework { get; private set; }
+        public string RuntimeIdentifier { get; private set; }
+        public string Name { get; private set; }
```

Comment:
This wouldn't be readonly anymore I believe. This should have a private readonly field, and then a public getter.